### ISSUE TITLE
Javascript syntax highlighting not working in 0.0.5?

### ISSUE THREAD

**Issue by CyberFerret** [davidcornu/zed-haml#6]
I just upgraded the HAML extension to 0.0.5 in my Zed editor today, and it seems that the highlighting for any `:javascript` blocks in my HAML files no longer has syntax highlighting?  (same colour for the entire block).  

I was sure it was working really well in 0.0.3 which was the previous version I had but seems to have stopped now.  I uninstalled and reinstalled the extension to be sure, but no change.

I haven't checked `:css` blocks to see if there is a problem there as well.

**Comment by vitallium** [comment]
@CyberFerret Hi, can you please provide a code sample?

**Comment by vitallium** [comment]
I am asking that because in all my tests, I see highlighting in HAML filters `javascript` and `css`:

![CleanShot 2024-11-07 at 14 17 51@2x](https://github.com/user-attachments/assets/292cfd04-adee-4096-9217-e31c4fb2ac5e)

![CleanShot 2024-11-07 at 14 18 04@2x](https://github.com/user-attachments/assets/1a455b70-6079-4a61-b440-1a8749ab7c44)

**Comment by davidcornu** [comment]
> I am asking that because in all my tests, I see highlighting in HAML filters `javascript` and `css`

Ditto. Couldn't reproduce the issue locally.

@CyberFerret if you open the command palette and search for `debug: open syntax tree view`, that may give you a clue about what's going on in the specific areas that aren't highlighting correctly.

**Comment by CyberFerret** [comment]
Thanks for the response and tips @vitallium and @davidcornu.  Here is a screenshot of one of the code blocks that isn't working:

<img width="1101" alt="CleanShot 2024-11-08 at 07 06 46@2x" src="https://github.com/user-attachments/assets/11e30ec3-255c-433f-894c-bb10c5eee06e">

As you can see, from line 40 onwards, the javascript isn't formatted.  Great tip about the syntax tree view, and when looking at it, it appears that it throws an error at 40:1 for some reason, then renders the rest of the code (javascript) as text.

<img width="558" alt="CleanShot 2024-11-08 at 07 08 09@2x" src="https://github.com/user-attachments/assets/78779122-7034-447e-ba1f-30d4a89479c2">

**Comment by vitallium** [comment]
@CyberFerret aha, Thanks for screenshots! @davidcornu released v0.0.6 with a fix for your case. At least I hope so. Can you please install it and check?

**Comment by CyberFerret** [comment]
@vitallium Just upgraded to 0.0.6 with the same results (and same error in the tree debug) 😕  I restarted Zed and did a complete uninstall/reinstall of the plugin to be sure, but no change unfortunately.

**Comment by vitallium** [comment]
@CyberFerret Ok, that's fine. Is it possible to share that HAML template publicly?

**Comment by KaNe23** [comment]
I seem to have the same issues, here is a little example that breaks the Javascript part:

```haml
#container
  %input

:javascript
  document.addEventListener("DOMContentLoaded", function() {
    const input = document.querySelector('input');

    input.addEventListener('click', function() {
      alert("You clicked on: " + this);
    });
  });
```

![20241211_11h50m55s_grim](https://github.com/user-attachments/assets/b95bf2f2-5740-4bef-9d7f-a6215fa764f3)

there is this little workaround to fix it by adding a comment in line 3:

![20241211_11h52m58s_grim](https://github.com/user-attachments/assets/b70a1230-37e4-405a-8ed4-419a9402077e)

Hope that helps.

**Comment by davidcornu** [comment]
@KaNe23 are you on the latest version? There were a few fixes in `v0.0.6` and `v0.0.7` specifically for filters.

Edit - I can reproduce on `v0.0.7`. Looks like the filter is unexpected

![CleanShot 2024-12-11 at 10 46 33@2x](https://github.com/user-attachments/assets/cf80bd4b-57df-4ce8-9bf4-4c0e0db81a76)

Edit - interestingly enough, removing the empty line on 3 fixes it.

**Comment by vitallium** [comment]
Yeah, I think there is a nasty bug when the previous node capture an extra newline (and the empty line) and that causes that error. I will have a look, thanks for providing the reproduction. I will add it to tests to avoid regressions.

**Comment by vitallium** [comment]
Okay, I think I've fixed that issue in https://github.com/vitallium/tree-sitter-haml/commit/ca002c14a373534affb32079dad6ca4a493451b2 There indeed was a nasty bug where the capturing node could capture a new line right after it:

```haml
#id
<--- NEW LINE HERE WILL BE CAPTURED BY THE PREVIOUS NODE
```

and that caused unpredictable syntax result for HAML templates. [I opened a PR](https://github.com/davidcornu/zed-haml/pull/7) that updates the grammar, @davidcornu @KaNe23 it would be nice if you can check that PR and see if it works on your end too. Thanks!

Here is the screenshot with the fix.

![CleanShot 2024-12-21 at 19 42 57@2x](https://github.com/user-attachments/assets/62efb6d7-31f4-49d0-858b-aeea34962b2e)
__

**Comment by davidcornu** [comment]
Fix for this was released: https://github.com/zed-industries/extensions/pull/1881
